test/test_utils.jl

```diff
@@ -294,3 +294,41 @@ module DummyRefShapes
         )
     end
 end
+
+############################################################
+# Inverse parametric mapping ξ = ϕ(x) for testing hessians #             
+############################################################
+function function_value_from_physical_coord(interpolation::Interpolation, cell_coordinates, X::Vec{dim,T}, ue) where {dim,T}
+    n_basefuncs = getnbasefunctions(interpolation)
+    scalar_ip = interpolation isa Ferrite.ScalarInterpolation ? interpolation : interpolation.ip
+    @assert length(ue) == n_basefuncs
+    ξ = MAPPING(scalar_ip, cell_coordinates, X)
```

Comment:
```suggestion
    ξ = Ferrite.find_local_coordinate(scalar_ip, cell_coordinates, X)
```
and relax its type to allow dual numbers in X (or see comment above), and delete the `MAPPING` method below?

Reply:
xref https://github.com/Ferrite-FEM/Ferrite.jl/pull/820

Reply:
Looking at 820 and here, I think `tol` and `maxiter` (and others in 820) should be keyword arguments with default values to that function, which would make it non-breaking and easier to read the code. 

@termi-official: Yea, would be good to finish #820 too - if you want to, please merge with master and I'll have time this week to do a proper review. Would also be nice if there is a test case that fails on master but that 820 solves! 

Adding the tol and maxiter keywords here or in 820 won't matter, should be an easy resolve in any case I think.

Reply:
For the tests, it might be better to use the numerical gradient via `FiniteDiff.jl`, since there are discussion in `ForwardDiff.jl` that might change it so that it won't work for iterative procedures in the future unless care is taken to extract only the value-part of the residual during tolerance checking (xref https://github.com/JuliaDiff/ForwardDiff.jl/pull/481 and https://github.com/JuliaDiff/ForwardDiff.jl/issues/480).